xds/internal/resolver/xds_resolver_test.go

```diff
@@ -857,12 +859,94 @@ func (s) TestXDSResolverWRR(t *testing.T) {
 		picks[clustermanager.GetPickedClusterForTesting(res.Context)]++
 		res.OnCommitted()
 	}
-	want := map[string]int{"A": 10, "B": 20}
+	want := map[string]int{"cluster:A": 10, "cluster:B": 20}
 	if !reflect.DeepEqual(picks, want) {
 		t.Errorf("picked clusters = %v; want %v", picks, want)
 	}
 }
 
+func init() {
+	balancer.Register(cspB{})
+}
+
+type cspB struct{}
+
+func (cspB) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
+	return nil
+}
+
+func (cspB) Name() string {
+	return "csp_experimental"
+}
+
+type cspConfig struct {
+	ArbitraryField string `json:"arbitrary_field"`
+}
+
+func TestXDSResolverClusterSpecifierPlugin(t *testing.T) {
```

Comment:
A quick 1-liner description of what this tests would be helpful.  E.g. `// Tests that cluster specifier plugins produce the correct service config.` or something SG.